controllers/apply-next/simple/fields.js

```diff
@@ -0,0 +1,682 @@
+        type: 'text',
+        attributes: { size: 25 },
+        isRequired: true,
+        schema: Joi.string().required(),
+        messages: {
+            base: { en: 'Enter a town or city' }
+        }
+    },
+    legalContactAddressCounty: {
+        name: `legal-contact-address-county`,
+        label: { en: 'County', cy: '' },
+        type: 'text',
+        attributes: { size: 25 },
+        isRequired: true,
+        schema: Joi.string().required(),
+        messages: {
+            base: { en: 'Enter a county' }
+        }
+    },
+    legalContactAddressPostcode: postcodeField({
+        name: 'legal-contact-address-postcode',
+        label: { en: 'Postcode', cy: '' }
+    }),
+    legalContactEmail: emailField({
+        name: 'legal-contact-email',
+        label: { en: 'Email', cy: '' },
+        explanation: { en: 'We’ll use this whenever we get in touch about the project', cy: '' }
+    }),
+    legalContactPhone: {
+        name: 'legal-contact-phone',
+        autocompleteName: 'tel',
+        type: 'tel',
+        size: 30,
+        label: { en: 'Contact number', cy: '' },
+        isRequired: true,
+        schema: Joi.number().required(),
+        messages: {
+            base: { en: 'Enter a phone number' }
+        }
+    },
+    legalContactCommunicationNeeds: {
+        name: 'legal-contact-communication-needs',
+        type: 'checkbox',
+        label: { en: 'Does this contact have any communication needs?', cy: '' },
+        options: [
+            { value: 'audiotape', label: { en: 'Audiotape', cy: '' } },
+            { value: 'braille', label: { en: 'Braille', cy: '' } },
+            { value: 'large-print', label: { en: 'Large print', cy: '' } }
+        ],
+        isRequired: false,
+        get schema() {
+            return Joi.array()
+                .items(Joi.string().valid(this.options.map(option => option.value)))
+                .single()
+                .optional();
+        },
+        messages: {
+            'any.allowOnly': { en: 'Choose from one of the options provided', cy: '' }
+        }
+    },
+    bankAccountName: {
+        name: 'bank-account-name',
+        label: { en: 'Name on the bank account', cy: '' },
+        explanation: { en: 'Name of your organisation as it appears on your bank statement', cy: '' },
+        type: 'text',
+        isRequired: true,
+        schema: Joi.string().required(),
+        messages: {
+            base: { en: 'Enter the name on the account', cy: '' }
+        }
+    },
+    bankSortCode: {
+        name: 'bank-sort-code',
+        label: { en: 'Sort code', cy: '' },
+        type: 'string',
+        attributes: {
+            size: 20
+        },
+        isRequired: true,
+        schema: Joi.string().required(),
+        messages: {
+            base: { en: 'Enter a sort-code', cy: '' }
+        }
+    },
+    bankAccountNumber: {
+        name: 'bank-account-number',
+        label: { en: 'Account number', cy: '' },
+        type: 'string',
```

Comment:
Back and forth on this! If we're requiring a number in the schema then it should be `type: number`, right? Happy that sort code can be a string (for now...) but this is always just a number.

Reply:
Data-wise it might be numeric-ish, but I disagree that number is the right input type. It's not a discreet value (like a price or quantity). The increment/decrement controls on an `input[type=number]` don't make sense for an account number. In terms of UI a normal text input feels the best fit.

Reply:
Fair enough! https://www.brucelawson.co.uk/2012/credit-card-bank-account-numbers-in-html5-forms/ bears you out.